torch/_dynamo/_trace_wrapped_higher_order_op.py

```diff
@@ -43,39 +44,49 @@
 # compiled autograd do we inline into the function.
 
 
-def trace_wrapped(*args, **kwargs):
+def trace_wrapped(*args: Any, **kwargs: Any) -> Any:
     with torch.no_grad():
         return _trace_wrapped_op(*args, **kwargs)
 
 
 class TraceWrapped(HigherOrderOperator):
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__("trace_wrapped")
 
-    def __call__(self, *args, **kwargs):
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
         return super().__call__(*args, **kwargs)
 
 
 # TODO(jansel): need to ensure this does not get DCEed
 _trace_wrapped_op = TraceWrapped()
 
 
-def _assert_meta(grad, size, stride, dtype):
+def _assert_meta(
+    grad: torch.Tensor,
+    size: Tuple[int, ...],
+    stride: Tuple[int, ...],
+    dtype: torch.dtype,
+) -> torch.Tensor:
     assert grad.size() == size, "size mismatch"
     assert grad.stride() == stride, "stride mismatch"
     assert grad.dtype == dtype, "dtype mismatch"
     return grad
 
 
 @_trace_wrapped_op.py_impl(ProxyTorchDispatchMode)
-def inner_trace(mode, *args, bw_state=None, **kwargs):
-    def self_invoke(*args, **dyn_kwargs):
+def inner_trace(
+    mode: ProxyTorchDispatchMode,
+    *args: Any,
+    bw_state: Optional[BackwardState] = None,
+    **kwargs: Any,
+) -> Any:
+    def self_invoke(*args: Any, **dyn_kwargs: Any) -> Any:
         with torch.no_grad():
             return _trace_wrapped_op(*args, **dyn_kwargs, **kwargs)
 
-    def unwrap_proxies(x):
+    def unwrap_proxies(x: Any) -> Any:
         if isinstance(x, torch.Tensor):
-            return mode.tracer.unwrap_proxy(x)
+            return mode.tracer.unwrap_proxy(x)  # type: ignore[union-attr]
```

Comment:
Could be fixed an assert isinstance mode.tracer? / asserting mode.tracer is not None, likely.

Reply:
Alas not

```
>>> Lint for torch/_dynamo/_trace_wrapped_higher_order_op.py:

  Error (MYPY) [union-attr]
    Item "_GraphAppendingTracerEx" of "PythonKeyTracer |
    _GraphAppendingTracerEx" has no attribute "unwrap_proxy"

         87  |    def unwrap_proxies(x: Any) -> Any:
         88  |        if isinstance(x, torch.Tensor):
         89  |            assert mode.tracer is not None
    >>>  90  |            return mode.tracer.unwrap_proxy(x)
         91  |        if isinstance(x, (list, tuple)):
         92  |            return type(x)(map(unwrap_proxies, x))
         93  |        if x is None:
```